### PR TITLE
Check out the incredible enhancements I've crafted for our platform. This pull request is a game-changer, focusing on refining the user journey and optimizing performance:

### PR DESCRIPTION
**Pull request by XRichardroidX**
**1️⃣ Email & Gmail Sign Up:**

**Seamlessly register with your email or Gmail for a smoother onboarding experience.
2️⃣ Streamlined Sign Up Process:**

**Bid farewell to phone sign-ups, simplifying the user registration flow.
3️⃣ Facebook Sign Up Refinement:**

**Addressed and improved the functionality of Facebook sign-up, ensuring a seamless connection.
4️⃣ Immersive Product Exploration:**

**Dive into a richer user experience with the ability to view products in intricate detail.
5️⃣ Full Image Post Unveiling:**

**Unleash the potential of visuals by enabling users to view full-sized image posts effortlessly.
6️⃣ Dynamic Comment Section:**

**Connect with the community like never before – the comment section is now seamlessly linked to the backend.
7️⃣ Optimized Client-Server Interaction:**

**Boosted performance by optimizing downloads between the client and server for a faster, more responsive platform.
8️⃣ Likes/Love Post Feature:**

*Introducing a new dimension to user engagement – express your appreciation with the likes/love post feature.
Get ready to revolutionize user interactions and elevate the overall platform experience! 🌟🌐*

**#FeatureEnhancements #UserExperienceRevamped #Optimization #PullRequestMasterpiece**

### CODE REVIEW

**Review by Victor-chiemerie — CHANGES_REQUESTED** [review]
# A very Excellent work man

You've really put in great hours of work into this project 👍 👍 , and I'm very sure that you'll be done with it in less time
Tho, there are some little changes that I feel you should make
Check them out in the comments below 👇

File: lib/needlinc/shared-pages/auth-pages/confirmNumber.dart
[the remark this answers is not in the document]
Comment (XRichardroidX) [reply lib/needlinc/shared-pages/auth-pages/confirmNumber.dart:1]: The phone number is required to create your profile as a bio-data, although it is not needed for user verification.

File: lib/needlinc/shared-pages/product-details.dart
[the remark this answers is not in the document]
Comment (XRichardroidX) [reply lib/needlinc/shared-pages/product-details.dart:1]: Yes, We'll work on that, I just want us to fulfill the top agenda for now.

File: lib/needlinc/shared-pages/help-action.dart
[the remark this answers is not in the document]
Comment (XRichardroidX) [reply lib/needlinc/shared-pages/help-action.dart:1]: It's a message that will be used in the future when there is an error in the user account, although it isn't useful for now, when updating the code-base in the future, such file will come in handy.

File: lib/needlinc/shared-pages/auth-pages/sign-up.dart [hunk 1]
```diff
@@ -169,132 +358,129 @@ class _SignupPageState extends State<SignupPage> {
...
+                            notLoading = true;
+                            emailController.clear();
+                            fullNameController.clear();
+                            userNameController.clear();
+                            passwordController.clear();
+                            confirmPassController.clear();
+                          }
+                          else{
+                            showSnackBar(context, "Add a photo for this account");
+                          }
```
[the remark this answers is not in the document]
Comment (XRichardroidX) [reply lib/needlinc/shared-pages/auth-pages/sign-up.dart:457]: It is necessary for every user to have a profile picture or a company logo, anyone who will take his/her account seriously must expect fame and popularity and such requires identification.

File: lib/needlinc/shared-pages/auth-pages/intro-page.dart [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SingleChildScrollView(
+        child: InkWell(
+          onTap: (){
+            Navigator.pushReplacement(
+                context, MaterialPageRoute(builder: (context) => const SignupPage()));
+          },
```
[the remark this answers is not in the document]
Comment (XRichardroidX) [reply lib/needlinc/shared-pages/auth-pages/intro-page.dart:43]: I will correct that immediately, thanks 👍...!

File: lib/needlinc/shared-pages/auth-pages/sign-up.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Victor-chiemerie) [reply lib/needlinc/shared-pages/auth-pages/sign-up.dart:457]: What of a user that just wants to hire a freelancer for a job and doesn't want much with the app

**Review by Victor-chiemerie — APPROVED** [review]

**Pull request merged**